### PR TITLE
ipatopologysegment: Fix documentation sections and agument spec

### PR DESCRIPTION
**Pull request by t-woerner**
ansible-test with ansible-2.14 is adding a lot of new tests to ensure that the documentation section and the agument spec is complete. Needed changes:

DOCUMENTATION section

- `type: str` needs to be set for string parameters
- `author` needs to be given with the github user also: `Name (@user)`

argument_spec

- `type="str"` needs to added to all string parameters

The `copyright` date is extended with `-2022`.

### CODE REVIEW

File: plugins/modules/ipatopologysegment.py [hunk 1]
```diff
@@ -39,29 +39,35 @@
...
   name:
     description: Topology segment name, unique identifier.
+    type: str
     required: false
     aliases: ["cn"]
   left:
     description: Left replication node - an IPA server
+    type: str
     aliases: ["leftnode"]
   right:
```
Comment (rjeffman) [inline plugins/modules/ipatopologysegment.py:54]: Should it have `required: false` ?

File: plugins/modules/ipatopologysegment.py [hunk 2]
```diff
@@ -39,29 +39,35 @@
...
+    type: str
     aliases: ["rightnode"]
   direction:
     description: The direction a segment will be reinitialized
+    type: str
     required: false
     choices: ["left-to-right", "right-to-left"]
   state:
     description: State to ensure
+    type: str
```
Comment (rjeffman) [inline plugins/modules/ipatopologysegment.py:65]: Should it have `required: false` ?

File: plugins/modules/ipatopologysegment.py [hunk 3]
```diff
@@ -39,29 +39,35 @@
...
     description: Topology suffix
+    type: str
     required: true
     choices: ["domain", "ca", "domain+ca"]
   name:
     description: Topology segment name, unique identifier.
+    type: str
     required: false
     aliases: ["cn"]
   left:
```
Comment (rjeffman) [inline plugins/modules/ipatopologysegment.py:50]: Should it have `required: false` ?

File: plugins/modules/ipatopologysegment.py (hunk 1 above)
Comment (t-woerner) [reply plugins/modules/ipatopologysegment.py:55]: Done

File: plugins/modules/ipatopologysegment.py (hunk 3 above)
Comment (t-woerner) [reply plugins/modules/ipatopologysegment.py:50]: Done

**Review by rjeffman — APPROVED** [review]
LGTM

**Pull request merged**